src/iam/iam-base.ts

```diff
@@ -314,22 +335,23 @@ export class IAMBase {
   on(event: "accountChanged" | "networkChanged" | "disconnected", eventHandler: () => void) {
     if (!this._providerType) return;
     const isMetamask = this._providerType === WalletProvider.MetaMask;
+    const isEKC = this._providerType === WalletProvider.EKC;
     switch (event) {
       case "accountChanged": {
         isMetamask
           ? this._metamaskProvider?.on("accountsChanged", eventHandler)
-          : this._walletConnectService.getProvider().wc.on("session_update", eventHandler);
+          : !isEKC && this._walletConnectService.getProvider().wc.on("session_update", eventHandler);
         break;
       }
       case "disconnected": {
-        isMetamask === false &&
+        isMetamask === false && !isEKC &&
           this._walletConnectService.getProvider()?.wc.on("disconnect", eventHandler);
         break;
       }
       case "networkChanged": {
         isMetamask
           ? this._metamaskProvider?.on("networkChanged", eventHandler)
-          : this._walletConnectService.getProvider().wc.on("session_update", eventHandler);
+          : !isEKC && this._walletConnectService.getProvider().wc.on("session_update", eventHandler);
```

Comment:
Why `EKC` doesn't handle network events? Without this connection will stay close